### PR TITLE
fix compatibility issues against latest eap

### PR DESCRIPTION
**Pull request by yonip23**
### What is broken
The latest EAP is moved from log4j based logger to `java.util.logging` logger. We are using log4j appender to send error reports to our services.
This means that in the new EAP not only does the appender fails to compile, it wouldn't do anything at all - because no log4j logger is running.

### The fix
There are a few approaches, but I chose to "hack" IJ's plugin SDK logging system and add our logic before each call to the logger.
Because the api is not built for this use case, there's a bit of an acrobatic code there in the package `logging`, but not too much imo.

### CODE REVIEW

**Comment by yonip23** [comment]
lateste eap compatibility check on this branch passes: https://github.com/codota/tabnine-intellij/runs/5189478567?check_suite_focus=true

File: src/main/java/com/tabnine/logging/TabnineLogDispatcher.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    @Throws(Exception::class)
+    fun dispatchLog(level: String, message: String, stackTrace: String?) {
+        ApplicationManager.getApplication()
+            .executeOnPooledThread {
+                val postRequest = HttpPost(String.format("%s/logs/%s", Config.LOGGER_HOST, level))
+                val requestBody = baseRequestBody.deepCopy()
+                requestBody.addProperty("message", message)
+                // requestBody.addProperty("stackTrace", stackTrace);
```
Comment (boaz-codota) [inline src/main/java/com/tabnine/logging/TabnineLogDispatcher.kt:36]: no leaving comments behind

**Review by boaz-codota — APPROVED** [review]
LGTM

**Comment by boaz-codota** [comment]
BUT: I think you could add testing against the latest heap in the CI @yonip23

**Comment by yonip23** [comment]
> BUT: I think you could add testing against the latest heap in the CI @yonip23 . Or is it already happening at https://github.com/codota/tabnine-intellij/blob/master/.github/workflows/compatibility-eap.yml#L10 ?

The latest eap job runs daily, like the nightly check we've got

**Comment by yonip23** [comment]
> lateste eap compatibility check on this branch passes: https://github.com/codota/tabnine-intellij/runs/5189478567?check_suite_focus=true

This is the job @boaz-codota

File: src/main/java/com/tabnine/logging/TabnineLogDispatcher.kt (hunk 1 above)
Comment (yonip23) [reply src/main/java/com/tabnine/logging/TabnineLogDispatcher.kt:36]: @yairco1990 im removing it then

**Review by amircodota — APPROVED** [review]
LGTM

**Review by yairco1990 — APPROVED** [review]
lgtm

**Pull request merged**